php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -42,6 +77,95 @@ public ExpressionTree expression(InternalSyntaxToken token) {
+    return new ExpandableStringCharactersTreeImpl(token);
+  }
+
+  public VariableIdentifierTree expandableStringVariableIdentifier(InternalSyntaxToken token) {
+    return new VariableIdentifierTreeImpl(new IdentifierTreeImpl(token));
+  }
+
+  public IdentifierTree identifier(InternalSyntaxToken token) {
+    return new IdentifierTreeImpl(token);
+  }
+
+  public ArrayAccessTree expandableArrayAccess(InternalSyntaxToken openBracket, ExpressionTree offset, InternalSyntaxToken closeBracket) {
+    return new ArrayAccessTreeImpl(openBracket, offset, closeBracket);
+  }
+
+  public MemberAccessTree expandableObjectMemberAccess(InternalSyntaxToken arrow, IdentifierTree property) {
+    return new MemberAccessTreeImpl(arrow, property);
+  }
+
+  public ExpressionTree encapsulatedSimpleVar(VariableIdentifierTree variableIdentifier, Optional<ExpressionTree> partial) {
+    if (partial.isPresent()) {
+
+      if (partial.get() instanceof ArrayAccessTree) {
+        ((ArrayAccessTreeImpl) partial.get()).complete(variableIdentifier);
+      } else {
+        ((MemberAccessTreeImpl) partial.get()).complete(variableIdentifier);
+      }
+      return partial.get();
+    }
+
+    return variableIdentifier;
+  }
+
+  public ExpressionTree expressionRecovery(InternalSyntaxToken token) {
+    return new IdentifierTreeImpl(token);
+  }
+
+  public ExpressionTree encapsulatedSemiComplexVariable(InternalSyntaxToken openDollarCurly, ExpressionTree expressionTree, InternalSyntaxToken closeCurly) {
+    return new CompoundVariableTreeImpl(openDollarCurly, expressionTree, closeCurly);
+  }
+
+  public VariableIdentifierTree encapsulatedVariableIdentifier(InternalSyntaxToken spaces, InternalSyntaxToken variableIdentifier) {
+    return new VariableIdentifierTreeImpl(new IdentifierTreeImpl(variableIdentifier));
+  }
+
+  public ExpressionTree encapsulatedComplexVariable(InternalSyntaxToken openCurly, Tree lookahead, ExpressionTree expression, InternalSyntaxToken closeCurly) {
+    return new ComputedVariableTreeImpl(openCurly, expression, closeCurly);
+  }
+
+  public ExpandableStringLiteralTree expandableStringLiteral(Tree spacing, InternalSyntaxToken openDoubleQuote, List<ExpressionTree> expressions, InternalSyntaxToken closeDoubleQuote) {
+    return new ExpandableStringLiteralTreeImpl(openDoubleQuote, expressions, closeDoubleQuote);
+  }
+
+  public YieldExpressionTree yieldExpression(InternalSyntaxToken yieldToken, ExpressionTree expr1, Optional<Tuple<InternalSyntaxToken, ExpressionTree>> expr2) {
+    if (expr2.isPresent()) {
+      return new YieldExpressionTreeImpl(yieldToken, expr1, expr2.get().first(), expr2.get().second());
+    }
+    return new YieldExpressionTreeImpl(yieldToken, expr1);
+  }
+
+  public ParenthesisedExpressionTree parenthesizedExpression(InternalSyntaxToken openParenthesis, ExpressionTree expression, InternalSyntaxToken closeParenthesis) {
+   return new ParenthesizedExpressionTreeImpl(openParenthesis, expression, closeParenthesis);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)